security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -220,9 +209,31 @@ func (sc *SecretManagerClient) CallUpdateCallback(resourceName string) {
 	}
 }
 
-// GenerateSecret generates new secret and cache the secret, this function is called by SDS.StreamSecrets
-// and SDS.FetchSecret. Since credential passing from client may change, regenerate secret every time
-// instead of reading from cache.
+// getCachedSecret: retrieve cached Secret Item (workload-certificate/workload-root) from secretManager client
+func (sc *SecretManagerClient) getCachedSecret(resourceName string) (secret *security.SecretItem) {
+	var rootCertBundle []byte
+	if c := sc.cache.GetWorkload(); c != nil {
+
+		cacheLog.WithLabels("ttl", time.Until(c.ExpireTime)).Info("returned workload certificate from cache")
+		if resourceName == security.RootCertReqResourceName {
+			rootCertBundle = sc.mergeConfigTrustBundle(c.RootCert)
+		} else {
+			rootCertBundle = c.RootCert
+		}
+		ns := &security.SecretItem{
+			ResourceName:     resourceName,
+			CertificateChain: c.CertificateChain,
```

Comment:
This can be fixed, but the function that passes this info to envoy now distinguishes the request based on the ResourceName and not on whether any of the parameter items are included in the request

Reply:
have fixed this.